man/systemd.network.xml

```diff
@@ -2525,6 +2525,17 @@ NFTSet=prefix:netdev:filter:eth_ipv4_prefix</programlisting>
         </listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>BOOTP=</varname></term>
+        <listitem>
+          <para>Takes a boolean. The DHCPv4 client can be configured to communicate with BOOP servers that
+          don't accept Option 53, DHCP Message Type. In this configuration, a BOOTP Request is sent without
+          any options by default. A BOOTP reply that contains Option 1: Subnet Mask is expected.</para>
```

Comment:
Before I forget I'm not sure the subnet mask should be mentioned explicitly. networkd tries to guess it in
https://github.com/systemd/systemd/blob/cf8fd7148cd8fbdb79381202ce8686eed1de09d2/src/libsystemd-network/sd-dhcp-client.c#L1570-L1573
and `bootpc` tries to guess it as well. It's true that it can be wrong and it's better to supply the subnet mask explicitly technically both networkd and bootpc can handle that more or less gracefully. (I think it would be nice if networkd logged its guess to make it clear that it's a ~~wild~~ guess but it's unrelated to this PR)

Reply:
I agree the subnet mask should not be described as required/expected. It is probably rare that one wouldn't be provided, but the class-full approach that is already implemented is appropriate. I would not call it a guess, although I agree that it is worth logging that a default subnet mask was computed. 

Slightly off-topic, I would also like to suggest that when we reject a reply b/c it "lacks address, server or lease," it would be helpful to explicitly log which are missing. It is quite difficult from a user standpoint to determine what the client considers "missing." For example, we tried using the `FallbackLeaseLifetimeSec=infinity` option, and could not tell why we were still getting that error message. I had to walk the source code to figure out that it's b/c that option is not applied to DHCP_ACKs, and that it was not due to the lack of `siaddr` which at one point `bootpd` had stopped providing.

Reply:
> Slightly off-topic, I would also like to suggest that when we reject a reply b/c it "lacks address, server or lease," it would be helpful to explicitly log which are missing

Personally I agree that it would be helpful. I think it should be possible to open a new issue (which should get the "needs-better-log-message" label) to prevent it from getting lost here.